asreview/webapp/api/projects.py

```diff
@@ -214,6 +214,29 @@ def api_init_project():  # noqa: F401
     return jsonify(project.config), 201
 
 
+@bp.route("/dataset_readers", methods=["GET"])
+@login_required
+def api_list_data_readers():
+    """Get the list of available data readers and read formats."""
+    readers = list_readers()
+    unique_readers = []
+    unique_names = set([])
+    for reader in readers:
+        if reader.name not in unique_names:
+            unique_names.add(reader.name)
+            unique_readers.append(reader)
```

Comment:
How does this work? I assume that all readers are unique in `readers`?

We know that there are custom readers out there in the wild. They don't have name and label. Can't we use the entry_point name such that we remain backward compatible? Label is optional then.

Reply:
`list_readers` gives the readers as defined in `setup.py`. There we have defined the current readers as:
```
        "asreview.readers": [
            ".csv = asreview.io:CSVReader",
            ".tab = asreview.io:CSVReader",
            ".tsv = asreview.io:CSVReader",
            ".ris = asreview.io:RISReader",
            ".txt = asreview.io:RISReader",
            ".xlsx = asreview.io:ExcelReader",
        ],
```
So I'm basically just grouping this list by the class. Maybe it would be better to use the reader name instead of the extension in `setup.py`?

Reply:
I'm concerned that the code breaks for users with reader extensions (and there are some out there). The label and name are not available for these extensions. What about using the following code: 

```python
@bp.route("/dataset_readers", methods=["GET"])
@login_required
def api_list_data_readers():
    """Get the list of available data readers and read formats."""

    payload = {"result": []}
    for e in _entry_points(group="asreview.readers"):

        reader = e.load()
        label = reader.label if hasattr(reader, "label") else None

        payload["result"].append(
            {
                "name": e.name[1:],
                "suffix": e.name,
                "label": label,
            }
        )
    return jsonify(payload)
```

I have tested this, and it works when no label exists. The client side is easy to adapt to this. It's slightly different from what happens with the writers, but not too much imo.

Reply:
I guess it depends on if you want to get the available dataset readers, or if you want to get the accepted data formats. My code is more for the first, your code is more for the second. Right now we only use the accepted data formats, and in this case it's not really relevant what dataset reader class is used. So we could use your code. 

If someone makes a custom plugin that also reads `.txt` but parses it differently, you would need to be able to distinguish between the different reader classes.

Reply:
Fully agree. I don't have a favorite solution as long as it doesn't break custom readers.